File: Python/Algorithms/bubbleSort.py
'''
    @Author Jared Scott 
    This function implements the bubble sort algorithm. 
    This function will augment the list which is given as input
'''
def bubbleSort(inputList):
    didSwap = True
    count = 0
    while didSwap:
        didSwap = False
        for i in range(len(inputList) - 1):
            if inputList[i] > inputList[i + 1]:
                inputList[i], inputList[i + 1] = inputList[i + 1], inputList[i]
                count += 1
                print("Swapping...",inputList)
                didSwap = True
    print("Sorting finished\nThe Input List has been altered")
def main():
    list = [3,1,7,4,0,9,6]
    print("BEFORE:",list)
    bubbleSort(list) 
    print("AFTER:",list)

if __name__ == "__main__":
    main()

File: JavaScript/Algorithms/bubble_sort.js
/*
 *@Author Jared Scott ☯
 *This script is an implementation of the bubble sort algorithm for an array of integer values 
 *Passing an array of integers into this function will cause the array to be sorted in place (The input array is augmented) 
 */

function bubbleSort(Int_Array){
    //Main scan through items in the array 
    for(let i = 0; i < Int_Array.length; i++){
        //Secondary scan for comparison 
        for(let j = 0; j < Int_Array.length - i - 1; j++){
            //Comparing the two values from the scans 
            if(Int_Array[j + 1] < Int_Array[j]){
                //Swapping if needed 
                [Int_Array[j + 1],Int_Array[j]] = [Int_Array[j],Int_Array[j + 1]]
            }
        }
    };
    //The array has been sorted in place (the array has been altered)  
    return Int_Array;
};

console.log("Sorting the array of integers using a bubble sort. . .");
console.log("The array has been sorted: " + bubbleSort([5,3,8,4,6]));